### PR TITLE
Added a few methods to Quaternion and Vector3

### PR DESCRIPTION
**Pull request by BarthPaleologue**
I noticed there was a few useful methods not implemented regarding transformations of Vector3 by Quaternions.

# Here is what I added :

 ## Quaternion

- added ```subtractInPlace```
- added ```invertInPlace```
- added ```invert```
- added ```normalizeToNew```
- added ```lengthSquared```

## Vector3

- added ```applyRotationQuaternionToRef```
- added ```applyRotationQuaternionInPlace```
- added ```applyRotationQuaternion```

## Other

- Removed a few repetitions

### CODE REVIEW

**Comment by deltakosh** [comment]
Good job!

File: packages/dev/core/src/Maths/math.vector.ts [hunk 1]
```diff
@@ -1082,6 +1082,40 @@ export class Vector3 {
         return result.copyFromFloats(this._x * scale, this._y * scale, this._z * scale);
     }
 
+    /**
+     * Rotates the vector using the given unit quaternion and stores the new vector in result
+     * @param q the unit quaternion representing the rotation
+     * @param result the output vector
+     */
+    public applyRotationQuaternionToRef(q: Quaternion, result: Vector3): Vector3 {
```
Comment (sebavan) [inline packages/dev/core/src/Maths/math.vector.ts:1090]: Can you add @returns annotation in the three functions ?

Comment (BarthPaleologue) [reply packages/dev/core/src/Maths/math.vector.ts:1090]: Will do ! Don't know why it wasn't auto generated

Comment (BarthPaleologue) [reply packages/dev/core/src/Maths/math.vector.ts:1091]: All done!

**Pull request merged**